postgres/datadog_checks/postgres/metadata.py

```diff
@@ -466,10 +466,11 @@ def _collect_metadata_for_database(self, dbname):
 
     @tracked_method(agent_check_getter=agent_check_getter)
     def _collect_postgres_settings(self):
-        with self._check._get_main_db().cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
-            self._log.debug("Running query [%s]", PG_SETTINGS_QUERY)
-            self._time_since_last_settings_query = time.time()
-            cursor.execute(PG_SETTINGS_QUERY)
-            rows = cursor.fetchall()
-            self._log.debug("Loaded %s rows from pg_settings", len(rows))
-            return [dict(row) for row in rows]
+        with self._check._get_main_db() as conn:
```

Comment:
so now the main check thread will share a connection with all of the other threads? I'm a bit worried about the perf impact that could have

Reply:
oh nevermind, i didnt realize we were already sharing the main check thread with everything else. thats fine 👍